### PR TITLE
Document release procedures

### PR DESCRIPTION
**Pull request by pazner**
Add some documentation for the release procedure. For now I've added documentation for the Julia interface, @jedbrown can add notes for C, Python, and Rust.

### CODE REVIEW

**Comment by jedbrown** [comment]
Thanks for starting this. Having recently refactored the Rust crates and gone through with releasing them, here's an update to the process. We could include more testing of the packaged crates like I describe for Python.

Comments welcome, and rendered file here: https://github.com/CEED/libCEED/blob/releasing-notes/RELEASING.md

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+### Tagging and releasing on GitHub
+
+0. Confirm all the steps above, including all language bindings.
+1. `git commit -am'libCEED 0.8.1'`
+More frequently, this is amending the commit message on an in-progress commit, after rebasing if applicable on latest `main`.
+2. `git push` updates the PR holding release; opportunity for others to review
+3. `git switch main && git merge --ff-only HEAD@{1}` fast-forward merge into `main`
+4. `git tag --sign -m'libCEED 0.8.1'`
+5. `git push origin main v0.8.1`
+6. Draft a [new release on GitHub](https://github.com/CEED/libCEED/releases), using the couple sentences gathered for release notes.
```
Comment (jeremylt) [inline RELEASING.md:38]:
```suggestion
6. Draft a [new release on GitHub](https://github.com/CEED/libCEED/releases), using a few sentences gathered from the release notes.
```

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+Some minor bookkeeping updates are needed when releasing a new version of the core library.
+
+The version number must be updated in
+
+* `include/ceed/ceed.h`
+* `ceed.pc.template`
+* `Doxyfile`
+
+Additionally, the release notes in `doc/sphinx/source/releasenotes.rst` should be updated. Use `git log --first-parent v0.7..` to get a sense of the pull requests that have been merged and thus might warrant emphasizing in the release notes. While doing this, gather a couple sentences for key features to highlight on [GitHub releases](https://github.com/CEED/libCEED/releases). The "Current Main" heading needs to be named for the release.
```
Comment (valeriabarra) [inline RELEASING.md:17]: Should we add any reference here in updating the User Manual version? That can easily fall out of date (as it is now)

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### Quality contral and good citizenry
```
Comment (pazner) [inline RELEASING.md:19]:
```suggestion
### Quality control and good citizenry
```

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+ UnsafeArrays = "1"
+-libCEED_jll = "0.7"
++libCEED_jll = "0.8"
+```
+Once this change is merged into libCEED's `main` branch, the updated package version can be
+registered using the GitHub registrator bot by commenting on the commit:
+
+> @JuliaRegistrator register branch=main subdir=julia/LibCEED.jl
+
+At this point, the bot should create against the [general Julia
```
Comment (pazner) [inline RELEASING.md:109]:
```suggestion
At this point, the bot should create a PR against the [general Julia
```

File: RELEASING.md (hunk 2 above)
Comment (jedbrown) [reply RELEASING.md:17]: Updated now, and here's a PR for the users manual https://github.com/CEED/libCEED/pull/777

You can find the version history here.
https://zenodo.org/record/4895340

**Review by jedbrown — APPROVED** [review]

File: RELEASING.md (hunk 2 above)
Comment (valeriabarra) [reply RELEASING.md:17]: Thanks Jed!

**Review by jeremylt — APPROVED** [review]
LGTM

**Pull request merged**